frontend/src/metabase/admin/settings/selectors.js

```diff
@@ -84,6 +88,22 @@ const SECTIONS = updateSectionsWithPlugins({
         widget: SiteUrlWidget,
         warningMessage: t`Only change this if you know what you're doing!`,
       },
+      {
+        key: "custom-homepage",
+        display_name: t`Custom Homepage`,
+        type: "boolean",
+        requireUserUpdate: true,
```

Comment:
This is a very specific thing that makes the generic settings code aware of the user data reloading. It is already a mess but I'd suggest adding a bit of an abstraction here. Can we have a property like `getPostUpdateAction` that will call the `refreshCurrentUser` thunk?